Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -2781,6 +2783,41 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     Internals& internals() { return m_internals; }
     const Internals& internals() const { return m_internals; }
 
+    class ProcessActivityState {
+        WebPageProxy& m_page;
+
+        std::unique_ptr<ProcessThrottlerActivity> m_isVisibleActivity;
+#if PLATFORM(MAC)
+        ProcessThrottlerTimedActivity m_wasRecentlyVisibleActivity;
```

Comment:
Why aren't you using a std::unique_ptr here like the others? Would solve your problem.